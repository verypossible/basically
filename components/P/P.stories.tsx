import * as React from 'react'

import { text, withKnobs } from '@storybook/addon-knobs'
import { storiesOf } from '@storybook/react'
import info from '../../utils/storybook-info'
import P from './index'

const withInfo = info(`
### Notes

The notes need to be updated for the new component.

### Customization

\`\`\`
import { P } from 'basically'
import styled from 'styled-components'

const MyP = styled(P)\`
  background: blue;
  color: white;
\`;

<MyP>I'm Customized</MyP>
\`\`\`
`)

const stories = storiesOf('Components/P', module)
const paragraph = `
Bacon ipsum dolor amet pork landjaeger shankle bresaola short ribs. Flank strip steak drumstick kevin, buffalo brisket
sausage beef. Ball tip shoulder prosciutto pork belly meatball, ribeye picanha pork chop burgdoggen brisket frankfurter
ground round hamburger bacon. Pork loin tri-tip strip steak prosciutto ribeye flank shankle kielbasa cow brisket. Corned
beef pork shank shoulder meatball landjaeger leberkas turkey, jerky turducken ground round beef. Meatloaf landjaeger ham
turducken jowl cupim hamburger corned beef leberkas andouille shoulder meatball.
`
stories.addDecorator(withKnobs)

stories.add('Documentation', withInfo(() => {
  return <P>{text('children', paragraph)}</P>
}))
