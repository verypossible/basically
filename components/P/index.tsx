import styled from 'styled-components'

export const P = styled.p`
  font-size: 1rem;
  line-height: 1.4rem;
  color: #333;
  margin: 0.2rem 0;
`

export default P
