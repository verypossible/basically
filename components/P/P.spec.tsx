import * as React from 'react'

import { shallow } from 'enzyme'
import 'jest-styled-components'

import P from './index'

describe('P', () => {
  it('matches the snapshot', () => {
    const tree = shallow(<P>Hi There</P>)
    expect(tree).toMatchSnapshot()
  })
})
