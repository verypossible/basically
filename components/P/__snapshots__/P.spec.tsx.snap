// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`P matches the snapshot 1`] = `
.c0 {
  font-size: 1rem;
  line-height: 1.4rem;
  color: #333;
  margin: 0.2rem 0;
}

<p
  className="c0"
>
  Hi There
</p>
`;
